server/bin/pbench-verify-backup-tarballs.py

```diff
@@ -0,0 +1,358 @@
+    except BadConfig as e:
+        print("{}: {}".format(_NAME_, e), file=sys.stderr)
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+
+    archive = config.ARCHIVE
+    if not os.path.isdir(archive):
+        logger.error(
+            "The setting for ARCHIVE in the config file is {}, but that is not a directory".format(archive))
+        return 1
+
+    # add a BACKUP field to the config object
+    config.BACKUP = backup = config.conf.get("pbench-server", "pbench-backup-dir")
+    if len(backup) == 0:
+        logger.error(
+            "Unspecified backup directory, no pbench-backup-dir config in pbench-server section")
+        return 1
+
+    if not os.path.isdir(backup):
+        logger.error("The setting for BACKUP in the config file is {}, but that is not a directory".format(backup))
+        return 1
+
+    # call the s3config class
+    s3_config_obj = S3Config(config)
+    s3_config_obj = sanity_check(s3_config_obj, logger)
+
+    logger.info('start-{}'.format(config.TS))
+
+    prog = os.path.basename(sys.argv[0])
+
+    sts = 0
+    # N.B. tmpdir is the pathname of the temp directory.
+    with tempfile.TemporaryDirectory() as tmpdir:
+
+        archive_obj = BackupObject("archive", config.ARCHIVE)
+        local_backup_obj = BackupObject("backup", config.BACKUP)
+        s3_backup_obj = BackupObject("s3", s3_config_obj)
+
+        # Check the data integrity in Archive.
+        md5_result_archive = checkmd5(config.ARCHIVE, tmpdir, archive_obj, logger)
+
+        # Check the data integrity in Backup.
+        md5_result_backup = checkmd5(config.BACKUP, tmpdir, local_backup_obj, logger)
+
+        with tempfile.NamedTemporaryFile(mode='w+t', dir=tmpdir) as report:
+            report.write("{}.{}({})\n".format(prog, config.TS, config.PBENCH_ENV))
+
+            if s3_config_obj is None:
+                report.write('S3 backup service is inaccessible.')
+
+            if md5_result_archive > 0:
+                # create a report for failed md5 results from archive
+                report_failed_md5(archive_obj, tmpdir, report, logger)
+                sts += 1
+
+            if md5_result_backup > 0:
+                # create a report for failed md5 results from archive
+                report_failed_md5(local_backup_obj, tmpdir, report, logger)
+                sts += 1
+
+            if md5_result_archive <= md5_result_backup:
+                # Compare archive with s3 backup.
+                compare_result = compare_with_s3_backup(s3_config_obj,
+                                                        archive_obj,
+                                                        tmpdir, report, logger)
+            else:
+                # Compare local backup with s3 backup.
+                compare_result = compare_with_s3_backup(s3_config_obj,
+                                                        local_backup_obj,
+                                                        tmpdir, report, logger)
+
```

Comment:
This can be simplified: we always compare archive with s3, so no need for an if. I think the best thing to do here is to do the list initializations (all three of them) separately and then call the two-way comparison function twice here: once to compare archive to local backup and once to compare archive to s3 backup.